src/charm.py

```diff
@@ -231,9 +240,21 @@ def _oathkeeper_service_is_running(self) -> bool:
         return service.is_running()
 
     @property
-    def _forward_auth_config(self) -> ForwardAuthConfig:
+    def _scheme(self) -> str:
         scheme = "https" if self._is_tls_ready() and not self.config["dev"] else "http"
-        decisions_url = f"{scheme}://{self.app.name}.{self.model.name}.svc.cluster.local:{OATHKEEPER_API_PORT}/decisions"
+        return scheme
+
+    @property
+    def _public_endpoint(self) -> str:
+        public_endpoint = (
+            self.ingress.url
+            or f"{self._scheme}://{self.app.name}.{self.model.name}.svc.cluster.local:{OATHKEEPER_API_PORT}"
+        )
+        return public_endpoint
+
+    @property
+    def _forward_auth_config(self) -> ForwardAuthConfig:
+        decisions_url = f"{self._scheme}://{self.app.name}.{self.model.name}.svc.cluster.local:{OATHKEEPER_API_PORT}/decisions"
```

Comment:
suggestion(non-blocking): how about a property for building this URL since it's needed in more than one place?